design/0001-frr.md

```diff
@@ -0,0 +1,347 @@
+│     nodes     │  │                    │   Protocol    │    │               │
+│    <<Node>>   ├──┘                    │ <<interface>> │    └───────────────┘
+│               │                       │               │
+└───────────────┘                       └───────▲───────┘
+                                                │<<implements>>
+                                     ┌──────────┴─────────┐
+                                     │                    │
+                              ┌──────┴────────┐  ┌────────┴──────┐
+                              │     main::    │  │    main::     │
+                              │ L2_controller │  │ bgp_controller│
+                              │   <<class>>   │  │   <<class>>   │
+                              │               │  │               │
+                              └───────────────┘  └───────┬───────┘
+                                                         │
+                                                         │  <<use>>
+                                                 ┌───────▼───────┐
+                                                 │               │
+                                                 │    session    │
+                                                 │ <<interface>> │
+                                                 │               │
+                                                 └───────▲───────┘
+                                                         │ <<implement>>
+                                             ┌───────────┴─────────────┐
+                                             │                         │
+                                     ┌───────┴───────┐         ┌───────┴───────┐
+                                     │               │         │               │
+                                     │  bgp_metallb  │         │    bgp_frr    │
+                                     │  <<package>>  │         │  <<package>>  │
+                                     │               │         │               │
+                                     └───────────────┘         └───────┬───────┘
+                                                                       │
+                                                                       │
+                                                               ┌───────┴───────┐
+                                                               │               │
+                                                               │      frr      │
+                                                               │ <<container>> │
+                                                               │               │
+                                                               └───────────────┘
+
+```
+
+#### FRR deployment
+
+FRR will be deployed as a container. The speaker application will be configured
+at startup to use the FRR bgp daemon and will provide details for connection to
+the daemon (e.g. location of a socket)
+
```

Comment:
FRR will be deployed as daemonSet in the same name space as metallb-system right ?

Reply:
Yes, I think that would be the case.

Reply:
@msherif1234 I changed the test to suggest this.